### PR TITLE
resource: support resource expiration updates

### PR DESCRIPTION
**Pull request by grondo**
This PR adds support for the `expiration` key in the [RFC 28](https://flux-framework.readthedocs.io/projects/flux-rfc/en/latest/spec_28.html#additional-acquire-responses) "Additional acquire responses".

Presence of this key indicates that the expiration of the instance resource set has been updated.

This is built on top of #1104 since this PR adds to the sharness tests which is modified there.

### CODE REVIEW

**Comment by grondo** [comment]
Oh, and obviously the new tests here won't pass CI until flux-framework/flux-core#5522 is merged. Actually, perhaps I should make this test conditional on support for running job updates of duration.

**Comment by grondo** [comment]
Ok, that test now checks if running job updates are allowed and the test is skipped if not.

**Comment by grondo** [comment]
Rebased on updated testing in #1103

File: resource/modules/resource_match.cpp [hunk 1]
```diff
@@ -1236,13 +1236,15 @@ static void update_resource (flux_future_t *f, void *arg)
     int rc = -1;
```
Comment (garlick) [inline resource/modules/resource_match.cpp:1236]: Typo in commit message s/acquirej/acquire/

**Review by garlick — APPROVED** [review]
Not spotting any problems here except a trivial typo in a commit message.  LGTM

**Comment by grondo** [comment]
Thanks! I pushed a fix for the commit message typo and will set MWP.

**Comment by grondo** [comment]
Hm, the new test is hanging at shutdown occasionally in CI (i.e. it appears all tests have completed, but the test never exits). I'll see if I can reproduce this locally.

**Comment by grondo** [comment]
Ok, the hang was due to no scheduler loaded at the end of the test. I've fixed that and forced a push.

**Pull request merged**